### PR TITLE
Wehelp Stage2 Week1 Tasks

### PR DESCRIPTION
**Pull request by 54Laichu**
賴益楚

- GET Attractions API
http://54.157.160.232:8000/api/attractions?page=1
http://54.157.160.232:8000/api/attractions?page=0&keyword=劍潭
http://54.157.160.232:8000/api/attractions?page=0&keyword=北

- GET Attraction API
http://54.157.160.232:8000/api/attraction/10

- GET MRTs API
http://54.157.160.232:8000/api/mrts

### CODE REVIEW

**Review by cwpeng — CHANGES_REQUESTED** [review]
1. nextPage of the last page should be null, for example, http://54.157.160.232:8000/api/attractions?page=1&keyword=%E5%8C%97
2. response format of http://54.157.160.232:8000/api/mrts is incorrect.

File: app.py [hunk 1]
```diff
@@ -1,17 +1,109 @@
...
+        SELECT mrt
+        FROM attractions
+        GROUP BY mrt
+        ORDER BY COUNT(*) DESC
+        """)
+        mrt_tuple = cursor.fetchall()
+        cursor.close()
+        db.close()
+
+        mrt_list = [mrt[0] for mrt in mrt_tuple if mrt[0] is not None]
```
Comment (54Laichu) [inline app.py:101]: 已修正錯誤的資料格式

File: app.py [hunk 2]
```diff
@@ -1,17 +1,109 @@
...
+    params.extend([limit, offset])
+
+    try:
+        cursor.execute(query, tuple(params))
+        attractions = cursor.fetchall()
+
+        cursor.execute("SELECT COUNT(*) as total FROM attractions")
+        total_records = cursor.fetchone()["total"]
+        total_pages = (total_records + limit - 1) // limit
+        next_page = page + 1 if len(attractions) == limit and page + 1 < total_pages else None
```
Comment (54Laichu) [inline app.py:47]: 已修正錯誤的頁碼規則

**Review by cwpeng — APPROVED** [review]
OK. Done.

**Pull request merged**